main.tf

```diff
@@ -95,7 +95,7 @@ resource "azurerm_kubernetes_cluster" "main" {
     }
   }
   dynamic "azure_active_directory_role_based_access_control" {
-    for_each = var.role_based_access_control_enabled && var.rbac_aad_managed ? ["rbac"] : []
+    for_each = var.role_based_access_control_enabled && var.rbac_aad_azure_rbac_enabled && var.rbac_aad_managed ? ["rbac"] : []
```

Comment:
The only way that we can set an `azure_active_directory_role_based_access_control` block with `var.rbac_aad_managed = true` is the `azure_active_directory_role_based_access_control.azure_rbac_enabled = true`, we cannot set it to `null` or `false`, is that correct? Why the provider allows us to configure `azure_rbac_enabled`'s value?